### PR TITLE
chore: update observeQuery docs

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
_Issue #, if available:_

_Description of changes:_
- Update Observe Query docs with the goal of clarifying that it can be used for real-time use cases after sync has completed.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by offlineprogrammer — APPROVED** [review]

File: src/fragments/lib/datastore/native_common/real-time.mdx [hunk 1]
```diff
@@ -20,11 +20,15 @@ import flutter3 from "/src/fragments/lib/datastore/flutter/real-time/observe-sni
...
+The first snapshot returned from `observeQuery` will contain the same results as calling `query` directly on your Model - a collection of all the locally-available records. Subsequent snapshots will be emitted as updates to the dataset are received in the local store.
 
-`observeQuery` also accepts the same predicates and sorting options as `query`.
+While data is syncing from the cloud, snapshots will contain all of the items synced so far and a `isSynced` status of `false`. When the sync process is complete, a snapshot will be emitted with all the records in the local store and a `isSynced` status of `true`. 
+
+In addition to typical real-time use cases, `observeQuery` can be used on app launch to show your customers an initial data set from the local store as new data is syncing.
+
+`observeQuery` also accepts the same [predicates](/lib/datastore/data-access#predicates) and [sorting](/lib/datastore/data-access#sort) options as `query`.
```
Comment (Jordan-Nelson) [inline src/fragments/lib/datastore/native_common/real-time.mdx:31]: Note: these links will not work as expected until https://github.com/aws-amplify/docs/pull/3859 is merged.

File: src/fragments/lib/datastore/native_common/real-time.mdx [hunk 2]
```diff
@@ -20,11 +20,15 @@ import flutter3 from "/src/fragments/lib/datastore/flutter/real-time/observe-sni
...
 
-The first snapshot returned from `observeQuery` will contain the same results as calling `query` directly on your Model - a collection of all the locally-available records. Subsequent snapshots will contain all of the `items` synced so far and will continue sending snapshots until `isSynced` is `true`.
+The first snapshot returned from `observeQuery` will contain the same results as calling `query` directly on your Model - a collection of all the locally-available records. Subsequent snapshots will be emitted as updates to the dataset are received in the local store.
 
-`observeQuery` also accepts the same predicates and sorting options as `query`.
+While data is syncing from the cloud, snapshots will contain all of the items synced so far and a `isSynced` status of `false`. When the sync process is complete, a snapshot will be emitted with all the records in the local store and a `isSynced` status of `true`. 
+
+In addition to typical real-time use cases, `observeQuery` can be used on app launch to show your customers an initial data set from the local store as new data is syncing.
```
Comment (HuiSF) [inline src/fragments/lib/datastore/native_common/real-time.mdx:29]: >as new data is syncing

Could this be slightly clearer: "while new data is being synced from cloud."

**Comment by poojamat** [comment]
LGTM

File: src/fragments/lib/datastore/native_common/real-time.mdx (hunk 2 above)
Comment (nickarocho) [reply src/fragments/lib/datastore/native_common/real-time.mdx:29]: Good call, that sounds clearer to me too.

**Review by nickarocho — APPROVED** [review]
LGTM, this is way better 👍  Thanks Jordan!

File: src/fragments/lib/datastore/native_common/real-time.mdx (hunk 2 above)
Comment (Jordan-Nelson) [reply src/fragments/lib/datastore/native_common/real-time.mdx]: Agreed. Updated.

**Pull request merged**